hpx/parallel/util/numa_binding_allocator.hpp

```diff
@@ -630,9 +630,16 @@ namespace hpx { namespace compute { namespace host {
         }
 
     public:
+        // return the binding helper cast to a specific type
+        template <typename Binder>
+        std::shared_ptr<Binder> get_binding_helper_cast() const
+        {
+            return std::dynamic_pointer_cast<Binder>(binding_helper_);
+        }
```

Comment:
Where is this easier to use than just calling `std::dynamic_pointer_cast` directly?

Reply:
The fact that it's a one line function shouldn't be a reason for not having it in the place where it belongs. I could point to thousands of functions in HPX and say "Where is this easier to use than just writing the code yourself inline?". I use it in several places and it makes user code more readable (IMHO) than having std::dynamic_pointer_cast<long_name_in_here>. It's a convenience wrapper,nothing more.

Reply:
Fair enough. I was just asking since there's no code in here (e.g. in the NUMA allocator test) that uses it so I can't judge its usefulness myself.

Reply:
I remember why I did it now. There is no `get_binding_helper() ` function accessor in the allocator currently, so when a user needs to get it (it may be in a different scope from the original setup of the allocator), we'd need to add
```
 std::shared_ptr<Binder> get_binding_helper_cast() const
        {
            return binding_helper_;
        }
```
and the user would  need to then do
```
  std::shared_ptr<binder_type_2D> binder2 =         
      std::dynamic_pointer_cast<binder_type_2D>(a.localMatrix().get_allocator().get_binding_helper());
```
so rather than add a get_binder function all alone and then require the user to cast it, I just added the single one with the cast already in place. 

I think I'll keep it that way as it's a bit painful if the user has to cast it directly